coordinator/cql/src/main/java/org/apache/cassandra/stargate/metrics/ClientMetrics.java

```diff
@@ -327,6 +327,12 @@ public void markAuthError() {
       authError.increment();
     }
 
+    @Override
+    public void updateDriverInfo(ClientInfo clientInfo) {
+      Tags tagsByDriver = clientInfoTagProvider.getClientInfoTagsByDriver(clientInfo);
```

Comment:
the only thing that worries me here is that line `296` above.. since the tags can change now, if the `clientInfo` is `null` in the constructor, we would register a metric with the empty tags and then with non-empty tags.. I am not sure if this is a problem.. and if this situation can happen at all.. but I also see not too many options here to fix..